pkg/client/grafanacloud/client.go

```diff
@@ -0,0 +1,80 @@
+// Package grafanacloud provides an interface to the Grafana Cloud API.
+package grafanacloud
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"strings"
+
+	"gopkg.in/yaml.v2"
+)
+
+const integrationsAPIURL = "https://integrations-api.grafana.net"
+
+// Client is a grafanacloud API client.
+type Client struct {
+	c      *http.Client
+	apiKey string
+}
+
+// NewClient creates a new Grafana Cloud client. All requests made will be
+// performed using the provided http.Client c. If c is nil, the default
+// http client will be used instead.
+//
+// apiKey will be used to authenticate against the API.
+func NewClient(c *http.Client, apiKey string) *Client {
```

Comment:
What do you think about adding `NewClientFromEnv` that assumes a configured `GCLOUD_API_KEY`?

Reply:
I think the client code is too low level for environment variable logic. The caller can be responsible for retrieving environment variables, if necessary. 

(That being said, I don't really like environment variables for configuration, and I think having the flag here is enough given this is a hidden command we don't want to document)